internal/common/mqtt/mqttconfig.go

```diff
@@ -49,14 +49,10 @@ type Client struct {
 	protocol string
 }
 
-//Message is used to wrap the app id and payload into one and publish to broker
-type Message struct {
-	AppID   string
-	Payload string
-}
-
 var (
-	clientData map[string]*Client
+	publishData     map[string]*Client
```

Comment:
How about just set `publishData` as Client? I think we can publish various topic messages with a Client variable if we set `client ID` of broker to `device ID` of Edge Orchestration.

```suggestion
	publishData     Client
```

Reply:
If we make the clientID as deviceID of edgeorchestration, how will the message from different clients be distinguished? Another issue is then there will be only one connection of HomeEdge and not the Service Application since multiple connections from same ClientID(which in our case will be DeviceID) and hence connection will be again set and reset for any new client coming. 
PublishData is a HashMap storing the Client Object with clientID as key for ease and fast retrieval of client. Else the entire hashmap has to be scanned to find any client which will increase the time taken

Reply:
> If we make the clientID as deviceID of edgeorchestration, how will the message from different clients be distinguished? 

Is it necessary to make multiple connections to the broker when we distinguish which client's message it is? Instead of making multiple connections to the same topic, the client mark who sent it on the message.

>Another issue is then there will be only one connection of HomeEdge and not the Service Application since multiple connections from same ClientID(which in our case will be DeviceID) and hence connection will be again set and reset for any new client coming. 

Make sense. It looks good to have hash map structure. However I would like to manage `publishData` through the combination of `broker host` and `topic`.

Reply:
The connections were made not to diferentiate between the messages but between the clients. Broker takes just two vales, clientid and payload. So we cannot send more than two data and even if json is sent it will be taken as payload data. So if the same deviceID is sent as string, client distinction is not possible. So the design considered sending the appid of the application, that is requesting as client id to maintain uniqueness plus multiple apps of same client can request for different things. Hence is the design consideration in the current way